resources/qml/WelcomePages/AddPrinterBySelectionContent.qml

```diff
@@ -0,0 +1,154 @@
+            }
+        }
+
+        contentComponent: networkPrinterListComponent
+
+        Component
+        {
+            id: networkPrinterListComponent
+
+            AddNetworkPrinterScrollView
+            {
+                id: networkPrinterScrollView
+
+                maxItemCountAtOnce: 6  // show at max 6 items at once, otherwise you need to scroll.
+
+                onRefreshButtonClicked:
+                {
+                    UM.OutputDeviceManager.startDiscovery()
+                }
+
+                onAddByIpButtonClicked:
+                {
+                    base.gotoPage("add_printer_by_ip")
+                }
+            }
+        }
+    }
+
+    DropDownWidget
+    {
+        id: addLocalPrinterDropDown
+
+        anchors.top: addNetworkPrinterDropDown.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.margins: 20
+
+        title: catalog.i18nc("@label", "Add a non-networked printer")
+
+        onClicked:
+        {
+            if (contentShown)
+            {
+                addNetworkPrinterDropDown.contentShown = false
+            }
+        }
+
+        contentComponent: localPrinterListComponent
+
+        Component
+        {
+            id: localPrinterListComponent
+
+            AddLocalPrinterScrollView
+            {
+                id: localPrinterView
+
+                maxItemCountAtOnce: 10  // show at max 10 items at once, otherwise you need to scroll.
+            }
+        }
+    }
+
+    Cura.PrimaryButton
+    {
+        id: nextButton
+        anchors.right: parent.right
+        anchors.bottom: parent.bottom
+        anchors.margins: 40
+        enabled:
+        {
+            // If the network printer dropdown is expanded, make sure that there is a selected item
+            if (addNetworkPrinterDropDown.contentShown)
+            {
+                return addNetworkPrinterDropDown.contentItem.currentItem != null
+            }
+            else
+            {
+                return addLocalPrinterDropDown.contentItem.currentItem != null
+            }
+        }
+
+        text: catalog.i18nc("@button", "Next")
+        width: 140
+        fixedWidthMode: true
+        onClicked:
+        {
+            // Create a network printer or a local printer according to the selection
+            if (addNetworkPrinterDropDown.contentShown)
+            {
+                // Create a network printer
+                const networkPrinterItem = addNetworkPrinterDropDown.contentItem.currentItem
+                CuraApplication.getDiscoveredPrintersModel().createMachineFromDiscoveredPrinter(networkPrinterItem)
+            }
+            else
+            {
+                // Create a local printer
+                const localPrinterItem = addLocalPrinterDropDown.contentItem.currentItem
+                Cura.MachineManager.addMachine(localPrinterItem.id)
+            }
+
+            // TODO: implement machine actions
```

Comment:
TODO is implemented now, right?

Reply:
This is implemented in https://github.com/Ultimaker/Cura/pull/5470